akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/ibex35/Ibex35StatManager.scala

```diff
@@ -0,0 +1,264 @@
+  object UpdateStats //Called to Update stats (after Ibex35HistoricManager has inserted new data)
+  object CheckReadyToStart //Called to check if we have all the data needed to Start the manager
+  object StatsUpdatedSuccessfully //When stats has been generated succesfully
+}
+
+/**
+  * In charge of create stats on demand from the index
+  *
+  * The flow is: initial -> gettingInitialStatus -> waitingForDeletions -> ready -> updating -> ready...
+  *
+  */
+class Ibex35StatManager extends Actor with ActorLogging with Stash {
+
+  val esDAO =
+    context.actorOf(Props[Ibex35ESDAO], name = "Ibex35ESDAO_Manager")
+
+  override def receive: Receive = initial
+
+  def initial: Receive = {
+
+    case InitIbex35StatManager =>
+      log.info("Recovering initial status")
+
+      //Recover MM*** status
+      esDAO ! retrieveLastMMUpdated(StatsGenerator.MM200)
+      esDAO ! retrieveLastMMUpdated(StatsGenerator.MM100)
+      esDAO ! retrieveLastMMUpdated(StatsGenerator.MM40)
+      esDAO ! retrieveLastMMUpdated(StatsGenerator.MM20)
+      context.become(gettingInitialStatus((false, None), (false, None), (false, None), (false, None)))
+
+    case _ =>
+      stash()
+  }
+
+  type CheckedMM = (Boolean, Option[DateTime])
+
+  def gettingInitialStatus(
+                          lastMM200: CheckedMM,
+                          lastMM100: CheckedMM,
+                          lastMM40: CheckedMM,
+                          lastMM20: CheckedMM
+                          ): Receive = {
+
+    case DataRetrieved(_, _, data, rq) if rq.where.nonEmpty =>
+      rq.where.get match {
+        case TermQueryDefinition(Ibex35ESDAO.Stats.statsAttr, StatsGenerator.MM200._1, _, _) =>
+            context.become(
+                gettingInitialStatus(generateCheckedMM(data), lastMM100, lastMM40, lastMM20)
+              )
+
+        case TermQueryDefinition(Ibex35ESDAO.Stats.statsAttr, StatsGenerator.MM100._1, _, _) =>
+          context.become(
+            gettingInitialStatus(lastMM200, generateCheckedMM(data), lastMM40, lastMM20)
+          )
+
+        case TermQueryDefinition(Ibex35ESDAO.Stats.statsAttr, StatsGenerator.MM40._1, _, _) =>
+          context.become(
+            gettingInitialStatus(lastMM200, lastMM100, generateCheckedMM(data), lastMM20)
+          )
+
+        case TermQueryDefinition(Ibex35ESDAO.Stats.statsAttr, StatsGenerator.MM20._1, _, _) =>
+          context.become(
+            gettingInitialStatus(lastMM200, lastMM100, lastMM40, generateCheckedMM(data))
+          )
+
+        case _ => log.error("Unknown where to threat!!")
+      }
+      self ! CheckReadyToStart
+
+
+    case ErrorRetrievingData(ex, index, typeName, rq) =>
+      log.error(s"Can't get stat for init from $index / $typeName, retrying in 30 seconds")
+      context.system.scheduler.scheduleOnce(30 seconds, esDAO, rq)
+
+    case CheckReadyToStart =>
+      if(lastMM200._1 && lastMM100._1 && lastMM40._1 && lastMM20._1){
```

Comment:
Mmmm I would probably turn the ```CheckedMM``` type into a ```case class``` for making this part of code more readable.

Reply:
I mean, I always try to avoid using the tuple accessors (```._1```,```._2```,...)

Reply:
Yep...
I thought the same when I first wrote that... :)

Reply:
Done